Projects/Feature/HomeFeature/Sources/CategorySelect/CategorySelectCore.swift

```diff
@@ -16,23 +18,58 @@ import ComposableArchitecture
 public struct CategorySelectCore {
   @ObservableState
   public struct State: Equatable {
+    var selectType: CategorySelectType = .select
     var selectedCategory: PomodoroCategory?
-    var categoryList: [PomodoroCategory] = []
-    
+    var selectedEditCategory: PomodoroCategory?
+    var selectedDeleteCategory: [Int] = []
+    var categoryList: [PomodoroCategory] = [] {
+      didSet {
+        isCategoryAddAvailable = categoryList.count < 10
+      }
+    }
+    var isMenuViewShow: Bool = false
+    var isCategoryAddAvailable: Bool = true
     public init() {}
   }
   
   public enum Action {
     case onAppear
-    case dismissButtonTapped
-    case bottomCheckButtonTapped
-    
+    case showMenu(Bool)
+    case cancelButtonTapped
+    case editButtonTapped
+    case deleteButtonTapped
+
     case getCategoryListResponse(Result<[PomodoroCategory], Error>)
     
     case setSelectedCategory(PomodoroCategory?)
     case selectCategory(PomodoroCategory)
+    case selectEditCategory(PomodoroCategory)
+    case selectDeleteCategory(PomodoroCategory)
+
+    case addCategoryTapped
+    case deleteCategoriesTapped([Int])
   }
-  
+
+  public enum CategorySelectType {
+    case select, edit, delete
+
+    var title: String {
+      switch self {
+      case .select: return "카테고리"
+      case .edit: return "카테고리 수정"
+      case .delete: return "카테고리 삭제"
```

Comment:
case 싱글라인으로 쓸거면 return 키워드는 빼도 될거 같아
```suggestion
      case .select: "카테고리"
      case .edit: "카테고리 수정"
      case .delete: "카테고리 삭제"
```